### PR TITLE
Dependency update

### LINKED ISSUE #284: Update dependencies and enable dependabot.
Update dependencies and enable dependabot. 
I particularly want to update the consul-template dependency.

### PR DESCRIPTION
**Pull request by eikenb**
Updates all dependencies. First consul-template, with a couple minor fixes, and then everything else. And then dependabot.

Fixes #284

### CODE REVIEW

File: runner.go [hunk 1]
```diff
@@ -20,7 +20,6 @@ import (
 	"github.com/hashicorp/consul-template/config"
 	dep "github.com/hashicorp/consul-template/dependency"
 	"github.com/hashicorp/consul-template/watch"
-	shellwords "github.com/mattn/go-shellwords"
```
Comment (lornasong) [inline runner.go:23]: It looks like this might be the only import of `go-shellwords`. Suggestion to run a `go mod tidy` to clean this up

**Review by lornasong — APPROVED** [review]
makes sense to me!

File: runner.go (hunk 1 above)
Comment (eikenb) [reply runner.go:23]: I think I forgot to run a go mod tidy after the general update. I'm going to fix that and rebase.

Comment (eikenb) [reply runner.go:23]: Yep. that removed it. Thanks!

**Pull request merged**